build/scripts/restore.ps1

```diff
@@ -0,0 +1,26 @@
+param ([string]$nugetZipUrl = $(throw "Need an URL to the NuGet zip") )
+
+$destination = ${env:UserProfile}
+$outFilePath = [IO.Path]::ChangeExtension([IO.Path]::GetTempFileName(), "zip")
+write-host "Downloading $nugetZipUrl -> $outFilePath"
+$client = new-object System.Net.WebClient
+$client.DownloadFile($nugetZipUrl, $outFilePath)
+
+# It's possible for restore to run in parallel on the test machines.  As such
+# we need to restore only new files to handle simultaneous restore scenarios.
```

Comment:
@jaredpar, @Pilchie. A normal nuget restore (in v3 where it goes to the %UserProfile%\.nuget folder) there is no issue restoring from multiple builds simultaneously, as a restore does not delete or overwrite what already exists.

Provided this script can do the same (unzip without touching an existing file), then their should be no issue with concurrent restore.

Overall, parallel runs on the machine is the best option, because the machines can handle the runs and it is a more manageable solution to have 10 machines which can do 2 runs each, then it is to have 20 machines.